### PR TITLE
[AIRFLOW-3404] Add Amazon SES support

### PR DESCRIPTION
**Pull request by LoicGombeaud**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3404
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Similarly to `sendgrid.py`, `ses.py` defines an e-mail backend using Amazon SES.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
I don't know how to test this code, and would appreciate any support in that direction

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by rbs392** [comment]
👍

File: setup.py [hunk 1]
```diff
@@ -225,6 +225,7 @@ def write_version(filename=os.path.join(*['airflow',
 samba = ['pysmbclient>=0.1.3']
 segment = ['analytics-python>=1.2.9']
 sendgrid = ['sendgrid>=5.2.0']
+ses = ['boto3']
```
Comment (kaxil) [inline setup.py:228]: emr has `emr = ['boto3>=1.0.0, <1.8.0']` 

so please update with version range for ses as well

**Review by kaxil — CHANGES_REQUESTED** [review]
If possible, can you add some mock test???

File: setup.py (hunk 1 above)
Comment (LoicGombeaud) [reply setup.py:228]: s3 is even more restrictive: `['boto3>=1.7.0, <1.8.0']`, should I set both emr and ses to the same range as s3?

**Comment by Rotemlofer** [comment]
Any progress with this?

**Comment by kaxil** [comment]
@ashb Can you review this one please?

File: setup.py (hunk 1 above)
Comment (ashb) [reply setup.py:228]: Let's remove the dedicated extra option for `ses` - see #4524 for why.

File: airflow/contrib/utils/ses.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+def _ses_send_raw_email(message):
+        aws_region = configuration.get('ses', 'REGION')
+        ses = boto3.client('ses', region_name=aws_region)
+        data = message.as_string()
+        ses.send_raw_email(RawMessage={'Data': data})
+
+
+def _get_email_address_list(address_string):
```
Comment (ashb) [inline airflow/contrib/utils/ses.py:85]: We shouldn't need to duplicate this - -this function already exists elsewhere in the codebase we can reuse.

**Review by ashb — CHANGES_REQUESTED** [review]
Partial review only.

File: airflow/contrib/utils/ses.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    To use this plugin:
+    0. include ses subpackage as part of your Airflow installation, e.g.,
+    pip install apache-airflow[ses]
+    1. update [email] backend in airflow.cfg, i.e.,
+    [email]
+    email_backend = airflow.contrib.utils.ses.send_email
+    2. configure SES specific setting in airflow.cfg, e.g.:
+    [ses]
+    aws_region = eu-west-1
```
Comment (adamatrais) [inline airflow/contrib/utils/ses.py:45]: It should be region as you're calling:

        aws_region = configuration.get('ses', 'REGION')

**Comment by OmerJog** [comment]
@LoicAG are you still working on this PR?

**Comment by LoicGombeaud** [comment]
@OmerJog not at the moment, no; I might get back to it at some point, but can't guarantee it, sorry!

**Comment by georghildebrand** [comment]
Hi all, i am currently looking into how to use SES as mailing backend for airflow. Will there be any progress with this PR?

**Comment by georghildebrand** [comment]
@LoicAG haha, just realized that it was you who did the PR. Man i think this PR deserves it to be done.

**Pull request closed without merging**